python/lsst/pipe/tasks/calibrate.py

```diff
@@ -539,7 +539,7 @@ def calibrate(self, exposure, exposureIdInfo=None, background=None,
         # compute photometric calibration
         if self.config.doPhotoCal:
             try:
-                photoRes = self.photoCal.run(exposure, sourceCat=sourceCat)
+                photoRes = self.photoCal.run(exposure, sourceCat=sourceCat, expId=exposureIdInfo.expId)
                 exposure.getCalib().setFluxMag0(photoRes.calib.getFluxMag0())
```

Comment:
If no `exposureIdInfo` is passed in to `calibrate`, the expId here will evaluate to 0 due to 
```python
if exposureIdInfo is None:
     exposureIdInfo = ExposureIdInfo()
```

on line 486 and https://github.com/lsst/obs_base/blob/master/python/lsst/obs/base/exposureIdInfo.py#L53